lib/core/utils/preload.js

```diff
@@ -0,0 +1,136 @@
+/**
+ * Validated the preload object
+ * @param {Object | boolean} preload configuration object or boolean passed via the options parameter to axe.run
+ * @return {boolean}
+ * @private
+ */
+function isPreloadValidObject(preload) {
+	return (
+		typeof preload === 'object' &&
+		preload.hasOwnProperty('assets') &&
+		Array.isArray(preload.assets) &&
+		preload.assets.length
+	);
+}
+
+/**
+ * Returns a boolean which decides if preload is configured
+ * @param {Object} options run configuration options (or defaults) passed via axe.run
+ * @return {boolean}
+ */
+function shouldPreload(options) {
+	if (!options) {
+		return false;
+	}
+
+	if (!options.preload) {
+		return false;
+	}
+
+	if (typeof options.preload === typeof true) {
+		return options.preload;
+	}
+
+	if (isPreloadValidObject(options.preload)) {
+		return true;
+	}
+
+	return false;
+}
+axe.utils.shouldPreload = shouldPreload;
+
+/**
+ * Constructs a configuration object representing the preload requested assets & timeout
+ * @param {Object} options run configuration options (or defaults) passed via axe.run
+ * @return {Object}
+ */
+function getPreloadConfig(options) {
+	// default fallback configuration
+	let out = {
+		assets: axe.constants.preloadAssets,
+		timeout: axe.constants.preloadAssetsTimeout
+	};
+
+	// if type is boolean
+	if (typeof options.preload === typeof true) {
+		return out;
+	}
+
+	// if type is object - ensure an array of assets to load is specified
+	if (isPreloadValidObject(options.preload)) {
+		const requestedAssets = [];
+		options.preload.assets.forEach(asset => {
+			const a = asset.toLowerCase();
+			if (axe.constants.preloadAssets.includes(a)) {
+				// unique assets to load, in case user had requested same asset type many times.
+				if (!requestedAssets.includes(a)) {
+					requestedAssets.push(a);
+				}
+				return a;
+			} else {
+				const e =
+					`Requested asset: ${a}, not supported by aXe.` +
+					`Supported assets are: ${axe.constants.preloadAssets
+						.map(_ => _)
+						.join(', ')}.`;
+				throw new Error(e);
+			}
+		});
+		out.assets = requestedAssets;
+
+		if (options.preload.timeout) {
+			if (
+				typeof options.preload.timeout === 'number' &&
+				!Number.isNaN(options.preload.timeout)
+			) {
+				out.timeout = options.preload.timeout;
+			} else {
+				throw new Error(`preload timeout specified is not of type number`);
+			}
+		}
+
+		return out;
+	} else {
```

Comment:
This block is unnecessary, since we `return` on L92.

Reply:
Also, I suggest doing the inverse of this and de-denting the happy path above. For example:

```
if (!isPreloadValidObject(options.preload)) {
  throw new Error('No assets....')
}
```